languages/go/exercises/concept/structs/.docs/instructions.md

```diff
@@ -0,0 +1,55 @@
+In this exercise you'll be organizing races between various types of remote controlled cars. Each car has its own speed and battery drain characteristics.
+
+Cars start with full (100%) batteries. Each time you drive the car using the remote control, it covers the car's speed in meters and decreases the remaining battery percentage by its battery drain.
+
+If a car's battery is below its battery drain percentage, you can't drive the car anymore.
+
+Each race track has its own distance. Cars are tested by checking if they can finish the track without running out of battery.
+
+## 1. Creating a remote controlled car
+
+Allow creating a remote controller car by defining a function `NewCar` that takes the speed of the car in meters, and the battery drain percentage as its two parameters (both of type `int`) and returns a `Car` instance:
+
+```go
+speed := 5
+batteryDrain := 2
+car := NewCar(speed, batteryDrain)
+```
+
+## 2. Creating a race track
+
+Define another struct type called `Track` with the field `distance` of type integer.
+Allow creating a race track by defining a function `NewTrack` that takes the track's distance in meters as its sole parameter (which is of type `int`):
+
+```go
+distance := 800
+raceTrack := NewTrack(distance)
+```
+
+## 3. Drive the car
+
+Implement the `Drive` function that updates the number of meters driven based on the car's speed, and reduces the battery according to the battery drainage:
+
+```go
+speed := 5
+batteryDrain := 2
+car := NewCar(speed, batteryDrain)
+car = Drive(car)
```

Comment:
Just as a small check: does the `Drive` function also update the passed-in argument? Or does it return a brand new instance? If it doesn't return a new instance, there is perhaps something to be said for making `Drive` a void function (or the Go equivalent of that). But I don't know Go, so if this is not relevant, just ignore it :)

Reply:
In this case, car is not a pointer so it will not update the car given to the Drive function. `Pointers` and `methods` are not known at this point.

Reply:
👍 Sorry about my n00b Go questions :D

Reply:
No problem 😄